x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/CertGenUtils.java

```diff
@@ -270,13 +318,43 @@ private static String getDefaultSignatureAlgorithm(PrivateKey key) {
      */
     static PKCS10CertificationRequest generateCSR(KeyPair keyPair, X500Principal principal, GeneralNames sanList) throws IOException,
         OperatorCreationException {
+        return generateCSR(keyPair, principal, sanList, Set.of());
+    }
+
+    /**
+     * Generates a certificate signing request
+     *
+     * @param keyPair   the key pair that will be associated by the certificate generated from the certificate signing request
+     * @param principal the principal of the certificate; commonly referred to as the distinguished name (DN)
+     * @param sanList   the subject alternative names that should be added to the certificate as an X509v3 extension. May be
+     *                  {@code null}
+     * @param extendedKeyUsages the extended key usages that should be added to the certificate as an X509v3 extension. May be
+     *                  {@code null} or empty.
+     * @return a certificate signing request
+     */
+    static PKCS10CertificationRequest generateCSR(
+        KeyPair keyPair,
+        X500Principal principal,
+        GeneralNames sanList,
+        Set<ExtendedKeyUsage> extendedKeyUsages
+    ) throws IOException, OperatorCreationException {
         Objects.requireNonNull(keyPair, "Key-Pair must not be null");
         Objects.requireNonNull(keyPair.getPublic(), "Public-Key must not be null");
         Objects.requireNonNull(principal, "Principal must not be null");
         JcaPKCS10CertificationRequestBuilder builder = new JcaPKCS10CertificationRequestBuilder(principal, keyPair.getPublic());
+
+        ExtensionsGenerator extGen = new ExtensionsGenerator();
         if (sanList != null) {
-            ExtensionsGenerator extGen = new ExtensionsGenerator();
             extGen.addExtension(Extension.subjectAlternativeName, false, sanList);
+        }
+
+        if (extendedKeyUsages != null) {
```

Comment:
nit: The null check is redundant and ugly for collections.

Reply:
Agreed.